pkg/types/pod.go

```diff
@@ -54,22 +55,24 @@ func NewPodCache() PodCache {
 }
 
 func (p *podCache) Add(pod *corev1.Pod) error {
-	if pod.Status.Phase != corev1.PodRunning {
-		return ErrPodNotRunning
+	ip := pod.Status.PodIP
```

Comment:
Pod assigned ip does not mean that it is directly available. If the pod is not in the ready state, it will cause traffic to fail. The logic of health check is required, or the judgment is combined with the endpoint controller.

Reply:
It's not the responsibility of `podCache` to do it, also, it doesn't matter as we only use endpoints subset in the ready part of an Endpoints resource. So unready pod won't be referenced.

Reply:
Unready pod will be set to nodes in upstream, that is the problem.